src/components/NetworkAssetSelectInput/NetworkAssetSelectInput.tsx

```diff
@@ -305,11 +314,48 @@ const NetworkAssetSelectInput = ({
     walletAddress,
   ]);
 
+  useEffect(() => {
+    const handleBalanceGet = async () => {
+      if (!sdk || !walletAddress) return;
+      await getSmartWalletBalancesByChain(walletAddress, supportedChains);
```

Comment:
I'm thinking if these shouldn't go somewhere in the upper level like `EtherspotContextProvider` because everytime we will need balances we have to add these lines, right? wdyt

Reply:
done in fix/PRO-640 once we merge them we'll have this useEffect on the Context provider file